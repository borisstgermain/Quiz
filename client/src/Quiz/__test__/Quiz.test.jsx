import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import ConnectedQuiz, { Quiz, mapStateToProps } from '../Quiz';


describe('Quiz', () => {
  let wrapper;
  const questions = [{
    question: 'What is best language?',
    answer: 'JS',
  },
  {
    question: 'What is this?',
    answer: 'Kaka',
  }];
  const removeQuestion = jest.fn();
  const props = {
    questions,
    removeQuestion,
  };


  describe('Smoke tests', () => {
    test('Render', () => {
      shallow(<Quiz {...props} />);
    });
  });

  describe('Component testing', () => {
    beforeEach(() => {
      wrapper = shallow(<Quiz {...props} />);
    });

    test('Contains components', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('Props', () => {
      expect(wrapper.instance().props.questions).toEqual(questions);
    });

    test('State');
    test('Click hanler');
  });
});


describe('Connected Quiz', () => {
  describe('Smoke', () => {
    const mockStore = configureStore();
    const initialState = {
      quiz: {
        questions: [{
          question: 'What is best language?',
          answer: 'JS',
        },
        {
          question: 'What is this?',
          answer: 'Kaka',
        }],
        isFetching: '',
        error: '',
      },
    };
    const store = mockStore(initialState);

    test('Render without crashing', () => {
      shallow(<ConnectedQuiz store={store} />);
    });
  });

  describe('Component testing', () => {
    let mockStore;
    let store;
    let initialState;
    let wrapper;

    beforeEach(() => {
      mockStore = configureStore();
      initialState = {
        quiz: {
          questions: [{
            question: 'What is best language?',
            answer: 'JS',
          },
          {
            question: 'What is this?',
            answer: 'Kaka',
          }],
          isFetching: '',
          error: '',
        },
      };
      store = mockStore(initialState);
      wrapper = shallow(<ConnectedQuiz store={store} />);
    });


    test('Contains / Snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('Check Prop matches with initialState', () => {
      expect(wrapper.props().questions).toEqual(initialState.quiz.questions);
    });

    test('Correct MapStateToProps', () => {
      const expectedProps = {
        questions: [{
          question: 'What is best language?',
          answer: 'JS',
        },
        {
          question: 'What is this?',
          answer: 'Kaka',
        }],
        isFetching: '',
        error: '',
        hasQuestions: true,
      };
      const state = wrapper.instance().store.getState();

      expect(mapStateToProps(state)).toEqual(expectedProps);
    });
  });
});
