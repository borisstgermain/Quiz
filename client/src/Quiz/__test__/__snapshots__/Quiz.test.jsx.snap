// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected Quiz Component testing Contains / Snapshot 1`] = `
<Quiz
  error=""
  fetchQuestions={[Function]}
  hasQuestions={true}
  isFetching=""
  questions={
    Array [
      Object {
        "answer": "JS",
        "question": "What is best language?",
      },
      Object {
        "answer": "Kaka",
        "question": "What is this?",
      },
    ]
  }
  removeQuestion={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;

exports[`Quiz Component testing Contains components 1`] = `
<div
  className="quiz"
>
  <Question
    answer="JS"
    clickHandler={[Function]}
    isShown={false}
    question="What is best language?"
  />
  <button
    className="next-button  next-button--hide"
    onClick={[Function]}
  >
    Следующий →
  </button>
</div>
`;
